### PR TITLE
V2023.3 dependency updates

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I've reviewed the changes and run automated tests.

#### Why is this the best possible solution? Were any other approaches considered?
Switching to Android Gradle Plugin 8 required some changes so this is the biggest pr with dependency updates I have ever created. The biggest change is probably [Migrating to Non-Transitive R Classes](https://github.com/getodk/collect/pull/5658/commits/de3a1cf7bb9ecdd3b0e5b740b8701d2670049741) We could have disabled it for now but they claim it's better not too and probably at some point in the future we would be forced to do that either way see https://developer.android.com/build/optimize-your-build#use-non-transitive-r-classes

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It doesn't require testing.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/proguard-rules.txt [hunk 1]
```diff
@@ -5,6 +5,9 @@
 -dontwarn org.xmlpull.v1.**
 -dontwarn org.hamcrest.**
 -dontwarn com.rarepebble.**
+-dontwarn com.networknt.**
+-dontwarn android.content.res.**
+-dontwarn org.kxml2.io.**
```
Comment (seadowg) [inline collect_app/proguard-rules.txt:10]: I haven't had to touch ProGuard for a while. Remind me what `dontwarn` is doing here?

File: analytics/build.gradle [hunk 2]
```diff
@@ -26,8 +26,8 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
```
Comment (seadowg) [inline analytics/build.gradle:29]: We're ok to use Java 17 byte code on API 21? Does the new AGP take care of that?

Comment (grzesiek2010) [reply analytics/build.gradle:29]: Everything seems to be fine. I've installed the app on Android 5 and didn't get any errors.

File: collect_app/proguard-rules.txt (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/proguard-rules.txt:10]: It ignores some weird errors I got like:
`com.android.tools.r8.utils.b: Library class android.content.res.XmlResourceParser implements program class org.xmlpull.v1`
It's enough for building and installing the app but we might run into some bugs on runtime too. We will see.

**Review by seadowg — APPROVED** [review]

**Pull request merged**